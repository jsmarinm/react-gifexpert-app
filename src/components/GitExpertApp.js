//Functional component

import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GitGrid } from "./GitGrid";

const GitExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {

    //     // setCategories ([...categories, 'Splider Man']);
    //     setCategories ( cats => [...cats, 'Splider Man']);
    // };

    return(
        <>
            <h2>GitExpertApp</h2>
            <AddCategory setCategories= {setCategories} />
            <hr />
            {/* <button onClick={handleAdd} >Agregar </button> */}

            <ol>
                {
                    // categories.map(category => {
                    //     return <li key={ category }>{ category }</li>
                    // })
                    categories.map(category => (
                    <GitGrid 
                    key={category}
                    category={category} />
                    ))
                    
                 }
               
            </ol>
         
        </>
    );

};


export default GitExpertApp;