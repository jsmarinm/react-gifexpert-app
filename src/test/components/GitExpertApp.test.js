import React from 'react';
import { shallow } from 'enzyme';
import GitExpertApp from '../../components/GitExpertApp';


describe('Pruebas en componente GitExpertApp',() => {

 

    test('Mostrar componente', () => { 
        const wrap = shallow(<GitExpertApp />);
        expect(wrap).toMatchSnapshot();

     });

     test('debe mostrar una lista de categorias', () => { 

        const categories = ['Dragon ball', 'One punch'];
        const wrap = shallow(<GitExpertApp defaultCategories={categories} />);

        console.log(categories.length);
        expect(wrap).toMatchSnapshot();

        expect(wrap.find('GitGrid').length).toBe(categories.length);

      });

});



