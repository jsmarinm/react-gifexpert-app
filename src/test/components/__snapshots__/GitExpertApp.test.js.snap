// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GitExpertApp Mostrar componente 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente GitExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GitGrid
      category="Dragon ball"
      key="Dragon ball"
    />
    <GitGrid
      category="One punch"
      key="One punch"
    />
  </ol>
</Fragment>
`;
